inference-engine/src/cldnn_engine/ops/tensor_iterator.cpp

```diff
@@ -0,0 +1,200 @@
+// Copyright (C) 2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "cldnn_program.h"
+#include "cldnn_common_utils.h"
+#include "cldnn_engine.h"
+
+#include <cpp/ie_cnn_network.h>
+
+#include "ngraph/op/tensor_iterator.hpp"
+#include "ngraph/op/constant.hpp"
+#include "ngraph/op/util/sub_graph_base.hpp"
+
+#include "api/loop.hpp"
+#include "api/mutable_data.hpp"
+#include "api/reorder.hpp"
+
+#include <vector>
+#include <algorithm>
+
+using TensorIterator = ngraph::op::v0::TensorIterator;
+
+namespace CLDNNPlugin {
+
+static cldnn::mutable_data CreateIntData(Program &p, const cldnn::primitive_id& id, int32_t num) {
+    auto mem = cldnn::memory::allocate(p.GetEngine(),
+        { cldnn::data_types::i32, cldnn::format::bfyx, { 1, 1, 1, 1 } });
+    auto ptr = mem.pointer<int32_t>();
+    *ptr.begin() = num;
+    return {id, mem};
+}
+
+static cldnn::mutable_data CreateOutputData(Program &p, const std::shared_ptr<ngraph::Node>& op,
+                                            const cldnn::primitive_id& id, const cldnn::primitive_id& input,
```

Comment:
How about CreateAdditionalOutputData to present it is attempting to create output with idx >=1

Reply:
Agreed, changed from CreateOutputData to CreateAdditionalOutputData